kobo/apps/project_ownership/models/transfer.py

```diff
@@ -0,0 +1,328 @@
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Optional, Union
+
+from constance import config
+from django.conf import settings
+from django.db import models, transaction
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as t
+
+from kobo.apps.help.models import InAppMessage, InAppMessageUsers
+from kpi.constants import PERM_MANAGE_ASSET
+from kpi.deployment_backends.kc_access.utils import (
+    assign_applicable_kc_permissions,
+    kc_transaction_atomic,
+    reset_kc_permissions,
+)
+from kpi.fields import KpiUidField
+from kpi.models import Asset, ObjectPermission
+from .base import TimeStampedModel
+from .choices import TransferStatusChoices, TransferStatusTypeChoices
+from .invite import Invite
+from ..exceptions import TransferAlreadyProcessedException
+from ..tasks import async_task
+from ..utils import get_target_folder
+
+
+class Transfer(TimeStampedModel):
+
+    uid = KpiUidField(uid_prefix='pot')
+    asset = models.ForeignKey(
+        'kpi.Asset',
+        related_name='transfers',
+        on_delete=models.CASCADE,
+    )
+    invite = models.ForeignKey(
+        Invite,
+        related_name='transfers',
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        verbose_name = 'project ownership transfer'
+
+    def __str__(self) -> str:
+        return (
+            f'{self.asset}: '
+            f'{self.invite.sender.username} -> '
+            f'{self.invite.recipient.username}'
+        )
+
+    def process(self):
+        if self.status != TransferStatusChoices.PENDING.value:
+            raise TransferAlreadyProcessedException()
```

Comment:
The purpose of this specific exception is for logging purpose. If the transfer fails because of this, the name of the exception is saved in error field which makes debugging easier on failure.